pkg/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go

```diff
@@ -331,6 +331,12 @@ func (n *klusterletController) sync(ctx context.Context, controllerContext facto
 		}
 	}
 
+	// Cleanup some resources before apply
+	err = n.cleanBeforeApply(ctx, &config)
```

Comment:
is this temp for upgrade only in this release? I would add some doc here. Also maybe you should define an array on what should be removed and call delete directly?

Reply:
fixed